src/msg/msg_utils.rs

```diff
@@ -50,6 +50,30 @@ impl From<String> for TokenIdPair {
     }
 }
 
+// TODO:
+fn hash_order(_order: &crate::types::matchengine::messages::Order) -> String {
+    unimplemented!()
+}
+
+use crate::account::SignatureBJJ;
+use std::convert::TryInto;
+impl From<&matchengine::messages::Order> for crate::account::Signature {
```

Comment:
I think Order and Signature are two different things. 'From/Into' seems not to be good name. pub fn extractSignature(&Order) ?

Reply:
https://github.com/Fluidex/rollup-state-manager/pull/80/commits/727648de2cf9c372477f870fa95690acda3848e6